internal/common/function/function_test.go

```diff
@@ -115,36 +122,184 @@ func TestValidateBackupSpec(t *testing.T) {
-		namespace string
+func TestGenerateNacObjectNameWithUUID(t *testing.T) {
+	tests := []struct {
 		name      string
-		expected  string
+		namespace string
+		nabName   string
 	}{
 		{
-			namespace: "example-namespace",
-			name:      "example-name",
-			expected:  "nab-example-namespace-1cdadb729eaac1",
+			name:      "Valid names without truncation",
+			namespace: defaultStr,
+			nabName:   "my-backup",
+		},
+		{
+			name:      "Truncate nabName due to length",
+			namespace: "some",
+			nabName:   strings.Repeat("q", constant.MaximumNacObjectNameLength+10), // too long for DNS limit
+		},
+		{
+			name:      "Truncate very long namespace and very long name",
+			namespace: strings.Repeat("w", constant.MaximumNacObjectNameLength+10),
+			nabName:   strings.Repeat("e", constant.MaximumNacObjectNameLength+10),
+		},
+		{
+			name:      "nabName empty",
+			namespace: "example",
+			nabName:   constant.EmptyString,
+		},
+		{
+			name:      "namespace empty",
+			namespace: constant.EmptyString,
+			nabName:   "my-backup",
 		},
 		{
-			namespace: strings.Repeat("m", validation.DNS1123SubdomainMaxLength),
-			name:      "example-name",
-			expected:  fmt.Sprintf("nab-%s-1cdadb729eaac1", strings.Repeat("m", truncatedStringSize)),
+			name:      "very long name and namespace empty",
+			namespace: constant.EmptyString,
+			nabName:   strings.Repeat("r", constant.MaximumNacObjectNameLength+10),
+		},
+		{
+			name:      "very long namespace and name empty",
+			namespace: strings.Repeat("t", constant.MaximumNacObjectNameLength+10),
+			nabName:   constant.EmptyString,
+		},
+		{
+			name:      "empty namespace and empty name",
+			namespace: constant.EmptyString,
+			nabName:   constant.EmptyString,
 		},
 	}
 
-	for _, tc := range testCases {
-		actual := GenerateVeleroBackupName(tc.namespace, tc.name)
-		if actual != tc.expected {
-			t.Errorf("Expected: %s, but got: %s", tc.expected, actual)
-		}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GenerateNacObjectNameWithUUID(tt.namespace, tt.nabName)
+
+			// Check length, don't use constant.MaximumNacObjectNameLength here
+			// so if constant is changed test needs to be changed as well
+			if len(result) > validation.DNS1123LabelMaxLength {
+				t.Errorf("Generated name is too long: %s", result)
+			}
+
+			// Extract the last 36 characters, which should be the UUID
+			if len(result) < 36 {
+				t.Errorf("Generated name is too short to contain a valid UUID: %s", result)
+			}
+			uuidPart := result[len(result)-36:] // The UUID is always the last 36 characters
+
+			// Attempt to parse the UUID part
+			if _, err := uuid.Parse(uuidPart); err != nil {
+				t.Errorf("Last part is not a valid UUID: %s", uuidPart)
+			}
+
+			// Check if no double hyphens are present
+			if strings.Contains(result, "--") {
+				t.Errorf("Generated name contains double hyphens: %s", result)
+			}
+		})
+	}
+}
+
+func TestGetVeleroBackupByLabel(t *testing.T) {
```

Comment:
why not use testEnv instead of fake client?

Reply:
either way is fine imo